test/Resolver/DependencyResolverTest.php

```diff
@@ -261,4 +263,351 @@ public function testResolveConfiguredPreference(ConfigInterface $config, $reques
+            [ TestAsset\B::class, new TestAsset\ExtendedB(new TestAsset\A()), false ],
+            [ TestAsset\A::class, new TestAsset\A(), false ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideUsableParametersData
+     */
+    public function testUsableConfigParametersAreAccepted(string $type, $value, bool $builtin = false)
+    {
+        $class = uniqid('MockedTestClass');
+        $paramName = uniqid('param');
+        $definition = $this->mockDefintition([
+            $class => [
+                'parameters' => [
+                    $paramName => [
+                        'type' => $type,
+                        'builtin' => $builtin
+                    ],
+                ]
+            ]
+        ]);
+
+        $config = new Config([
+            'types' => [
+                $class => [
+                    'parameters' => [
+                        $paramName => $value
+                    ]
+                ]
+            ]
+        ]);
+
+        $resolver = new DependencyResolver($definition, $config);
+        $result = $resolver->resolveParameters($class);
+
+        $this->assertArrayHasKey($paramName, $result);
+        $this->assertInstanceOf(ValueInjection::class, $result[$paramName]);
+        $this->assertSame($value, $result[$paramName]->getValue());
+    }
+
+    /**
+     * Use Case:
+     *
+     * - A class requires an interface "A".
+     * - The configuration defines this parameter to inject another interface which extends "A"
+     *
+     * In this case the resolver must accept it.
+     */
+    public function testConfiguredExtendedInterfaceParameterSatisfiesRequiredInterfaceType()
+    {
+        $class = uniqid('MockedTestClass');
+        $paramName = uniqid('param');
+        $definition = $this->mockDefintition([
+            $class => [
+                'parameters' => [
+                    $paramName => [
+                        'type' => TestAsset\Hierarchy\InterfaceA::class
+                    ],
+                ]
+            ]
+        ]);
+
+        $config = new Config([
+            'types' => [
+                $class => [
+                    'parameters' => [
+                        $paramName => TestAsset\Hierarchy\InterfaceC::class
+                    ]
+                ]
+            ]
+        ]);
+
+        $resolver = new DependencyResolver($definition, $config);
+        $result = $resolver->resolveParameters($class);
+
+        $this->assertArrayHasKey($paramName, $result);
+        $this->assertInstanceOf(TypeInjection::class, $result[$paramName]);
+        $this->assertEquals(TestAsset\Hierarchy\InterfaceC::class, $result[$paramName]->getType());
+    }
+
+    public function provideIterableClassNames()
+    {
+        return [
+            [ TestAsset\Pseudotypes\IteratorImplementation::class ],
+            [ TestAsset\Pseudotypes\IteratorAggregateImplementation::class ],
+            [ \ArrayObject::class ],
+            [ \ArrayIterator::class ],
```

Comment:
- Add names to each case
- Import `ArrayObject` and `ArrayIterator` to this file